tests/e2e/system_test/test_reclaim_space_cronjob.py

```diff
@@ -0,0 +1,212 @@
+import logging
+import pytest
+import yaml
+from uuid import uuid4
+import os
+import time
+from tempfile import NamedTemporaryFile
+
+
+from ocs_ci.framework.testlib import (
+    skipif_ocp_version,
+    skipif_ocs_version,
+    ManageTest,
+    tier3,
+)
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.helpers.performance_lib import run_oc_command
+from ocs_ci.helpers import helpers
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import PVCNotCreated
+
+logger = logging.getLogger(__name__)
+ERRMSG = "Error in command"
+
+
+@tier3
+@skipif_ocp_version("<4.14")
+@skipif_ocs_version("<4.14")
+class TestReclaimSpaceCronJob(ManageTest):
+    """
+    Test that verifies creation of Reclaim Space Cron Jobs for RBD PCSs in openshift-* namespaces
+    The test also verifies that no Reclaim Space Cron Job is created for CephFS PVC in this namespace
+    """
+
+    def setup(self):
+        self.pvc_objs_created = []
+        self.temp_files_list = []
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        """
+        Cleanup the test environment
+
+        """
+
+        def finalizer():
+            # pvc objects usually are deleted in the test (and the second delete has no effect),
+            # if for some reason the pvcs were not deleted in the test - they will be deleted here
+            for pvc_obj in self.pvc_objs_created:
+                pvc_obj.delete()
+
+            for temp_file in self.temp_files_list:
+                if os.path.exists(temp_file):
+                    run_cmd(f"rm {temp_file}")
```

Comment:
Typically, finalizer is used to create any objects created as part of setup. 
Here, it is used to delete the PVCs. cant this be included in the test steps?

Reply:
In-case first delete successful won't the second delete fail the TC?

Reply:
Deletion of the PVCs is indeed executed as part of the regular test flow. 
However, for the cases where the test will fail for some reason and the PVCs will not be deleted - the finilizer will delete them so that we will not have any leftovers at the end. 

In case of the regular flow, when the PVCs are deleted as part of the test - the try to delete the PVCs as part of the finilizer is not time consuming, does not cause any errors and it will exit immediately.

Reply:
> In-case first delete successful won't the second delete fail the TC?

No, it won't fail the TC ( tested).  Is also won't consume time.

Reply:
@ypersky1980 can you trigger a run to check the code which is there in teardown? i.e. to validate in-case of PVC not deleted in 1st scenario and to clean it up in teardown/finalizer code

Reply:
I've commented out the pvc.delete() commands in the test and this is what I've got in the test run log: 

FAILED
------------------------------ live log teardown -------------------------------
00:53:12 - MainThread - test_auto_reclaim_space_cronjob - INFO  - Deleting pvc pvc-test-2834c992a22646978878c6e7a898171 in finalizer
00:53:12 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-22846a0eaf73419d8e35917ef0016385 delete PersistentVolumeClaim pvc-test-2834c992a22646978878c6e7a898171
00:53:13 - MainThread - test_auto_reclaim_space_cronjob - INFO  - Deleting pvc pvc-test-d3e3831763634f6e89d5d796d8f8c0d in finalizer
00:53:13 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-22846a0eaf73419d8e35917ef0016385 delete PersistentVolumeClaim pvc-test-d3e3831763634f6e89d5d796d8f8c0d
00:53:14 - MainThread - test_auto_reclaim_space_cronjob - INFO  - Deleting pvc pvc-test-7cab215f5bef4e1ab8468f69f6dae3d in finalizer
00:53:14 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-22846a0eaf73419d8e35917ef0016385 delete PersistentVolumeClaim pvc-test-7cab215f5bef4e1ab8468f69f6dae3d
00:53:15 - MainThread - test_auto_reclaim_space_cronjob - INFO  - Deleting pvc pvc-test-d9d2b2cdcf8649b2a49b8887221b101 in finalizer
00:53:15 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-22846a0eaf73419d8e35917ef0016385 delete PersistentVolumeClaim pvc-test-d9d2b2cdcf8649b2a49b8887221b101
00:53:16 - MainThread - test_auto_reclaim_space_cronjob - INFO  - Deleting pvc pvc-test-04ac03d8eb624045bab00cfb20bb78a in finalizer
00:53:16 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-22846a0eaf73419d8e35917ef0016385 delete PersistentVolumeClaim pvc-test-04ac03d8eb624045bab00cfb20bb78a
00:53:17 - MainThread - test_auto_reclaim_space_cronjob - INFO  - Deleting pvc pvc-test-49df79803e754754acaa71eada9815c in finalizer
00:53:17 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-22846a0eaf73419d8e35917ef0016385 delete PersistentVolumeClaim pvc-test-49df79803e754754acaa71eada9815c
00:53:19 - MainThread - test_auto_reclaim_space_cronjob - INFO  - Deleting pvc pvc-test-6cd78ec15d0f47758bd83e90dad7197 in finalizer
00:53:19 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-22846a0eaf73419d8e35917ef0016385 delete PersistentVolumeClaim pvc-test-6cd78ec15d0f47758bd83e90dad7197
00:53:20 - MainThread - test_auto_reclaim_space_cronjob - INFO  - Deleting pvc pvc-test-2cb377c089b14703ac8610895f131d5 in finalizer
00:53:20 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-22846a0eaf73419d8e35917ef0016385 delete PersistentVolumeClaim pvc-test-2cb377c089b14703ac8610895f131d5
00:53:21 - MainThread - test_auto_reclaim_space_cronjob - INFO  - Deleting pvc pvc-test-f74f7b6c6d8e4d44bfbb632a44cd50d in finalizer
00:53:21 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n openshift-22846a0eaf73419d8e35917ef0016385 delete PersistentVolumeClaim pvc-test-f74f7b6c6d8e4d44bfbb632a44cd50d
00:53:22 - MainThread - test_auto_reclaim_space_cronjob - INFO  - Deleting pvc pvc-test-75e667d1911e4596aaa48f554757a5f in finalizer